Realm/RLMRealm.mm

```diff
@@ -692,8 +692,13 @@ - (void)addOrUpdateObject:(RLMObject *)object {
     RLMAddObjectToRealm(object, self, true);
 }
 
-- (void)addOrUpdateObjectsFromArray:(id)array {
-    for (RLMObject *obj in array) {
+- (void)addOrUpdateObjects:(id<NSFastEnumeration>)objects {
+    for (RLMObject *obj in objects) {
+        if (![obj isKindOfClass:[RLMObject class]]) {
```

Comment:
These checks (including the pre-existing one in addObjects:) should be for RLMObjectBase.